Automattic/jetpack
# Calendly Block: Replace SubmitButton with new button block

Fixes #15711, Fixes #14564 

#### Changes proposed in this Pull Request:
* Replace the SubmitButton in the Calendly block with new shared Button block.
* Fixes minor bug where Calendly spinner for inline style widget isn't hidden when there are multiple Calendly blocks on the page.
* Switch to shared Button block also fixes issue where custom text color wasn't shown in editor.

#### Prerequisites:
- Calendly account to create blocks and embed codes
- Post with Calendly blocks created prior to applying this PR's changes.
  1.   Create a post.
  2.   Add two Calendly blocks using the `link` style.
  3.   Configure one of those blocks setting its various options e.g. background color.
  4.   Add one inline Calendly block. 
  5.   Save post.
- Have the AMP plugin installed and active

#### Testing instructions:

1. On the frontend, visit post containing previous versions of Calendly block.
2. Confirm the blocks display and function as they did when you set them up. 
    _Should still be using simple `<a>` markup._
3. Edit the post.
4. Confirm the Calendly blocks still appear correctly and their settings are correct.
5. Update the link style blocks using the new settings controls provided by button block.
6. Add new Calendly block using simple URL e.g. `https://calendly.com/<username>`
7. Add new Calendly block using customized embed code for a link style widget.
8. Check that the customized attributes from the embed code have been set on the button.
9. Save the post.
10. Re-visit the post on the frontend and confirm display and function of blocks.
11. Switch back to the editor and update the embed url under the Inspector Controls > Calender Settings with a link style URL containing custom colors.
12. Confirm button colors update, save and switch to the frontend view to check there as well.
13. Click the AMP link at the top of the page and confirm links work on the AMP request view.
  ( inline style block will show as a simple link )
14. Run tests in `extensions/blocks/calendly/test/utils.js`


#### Before:
<img width="1084" alt="Calendly-Before-SubmitButtonReplacement" src="https://user-images.githubusercontent.com/60436221/82451079-dbe0e900-9af0-11ea-90c7-fe15ff918514.png">

#### After:
<img width="1066" alt="Calendly-After-SubmitButtonReplacement" src="https://user-images.githubusercontent.com/60436221/82451104-e26f6080-9af0-11ea-9ee5-5ba85525ca92.png">


#### Proposed changelog entry for your changes:
* Update Calendly block to use new shared Button block.
